src/wms/src/main/java/org/geoserver/wms/map/DirectRasterRenderer.java

```diff
@@ -398,6 +402,7 @@ public RenderedImage render() throws FactoryException {
                 if (transparent && !image.getColorModel().hasAlpha()) {
                     image = addAlphaChannel(image);
                     worker.setImage(image);
+                    transparencyType = image.getColorModel().getTransferType();
```

Comment:
I'm wondering if that migth be an auto-complete issue.
Is actually getTransparencyType or is getTransferType correct, instead?

Reply:
Auto-complete issue, well spotted.

Reply:
Cool. beside of that, +1 to merge